packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -544,61 +633,55 @@ export function makeVirtualObjectManager(
           },
         });
       }
-      harden(wrappedState);
+      harden(state);
 
       if (initializing) {
         cache.remember(innerSelf);
       }
-      const self = actualize(wrappedState);
       let toHold;
       let toExpose;
-      const facetiousness = assessFacetiousness(self);
-      switch (facetiousness) {
-        case 'one': {
-          toHold = Far(tag, self);
-          vrm.checkOrAcquireFacetNames(kindID, null);
-          toExpose = toHold;
-          break;
-        }
-        case 'many': {
-          toExpose = {};
-          toHold = [];
-          const facetNames = Object.getOwnPropertyNames(self).sort();
-          assert(
-            facetNames.length > 1,
-            'a multi-facet object must have multiple facets',
-          );
-          vrm.checkOrAcquireFacetNames(kindID, facetNames);
-          for (const facetName of facetNames) {
-            const facet = Far(`${tag} ${facetName}`, self[facetName]);
-            toExpose[facetName] = facet;
-            toHold.push(facet);
-            facetToCohort.set(facet, toHold);
-          }
-          harden(toExpose);
-          break;
+      unweakable.add(state);
+      if (facetNames === null) {
+        const context = { state };
+        context.self = bindMethods(context, behaviorTemplate);
+        toHold = Far(tag, context.self);
+        // eslint-disable-next-line no-proto
+        linkToCohort.set(toHold.__proto__, toHold);
+        // eslint-disable-next-line no-proto
+        unweakable.add(toHold.__proto__);
+        toExpose = toHold;
+        harden(context);
+      } else {
+        toExpose = {};
+        toHold = [];
+        const facets = {};
+        const context = { state, facets };
+        for (const name of facetNames) {
+          facets[name] = bindMethods(context, behaviorTemplate[name]);
+          const facet = Far(`${tag} ${name}`, facets[name]);
+          // eslint-disable-next-line no-proto
+          linkToCohort.set(facet.__proto__, facet);
+          // eslint-disable-next-line no-proto
+          unweakable.add(facet.__proto__);
+          toExpose[name] = facet;
+          toHold.push(facet);
+          linkToCohort.set(facet, toHold);
         }
-        case 'not':
-          assert.fail(X`invalid self actualization for ${q(tag)}`);
-        default:
-          assert.fail(X`unexepected facetiousness: ${q(facetiousness)}`);
+        unweakable.add(facets);
+        harden(context);
+        harden(facets);
+        harden(toExpose);
```

Comment:
think we should `harden(toHold)` for completeness' sake?

Reply:
In practical terms that would result in neither harm nor benefit, but I like the aesthetic of positivism w.r.t immutability.